### ISSUE TITLE
An input can be a descendant of a element if input is inert

### ISSUE THREAD

**Issue by tkrotoff** [validator/validator#1904]
`Error: The element input must not appear as a descendant of the a element` should not appear if the descendant has the `inert` attribute.

Code to reproduce error (https://validator.w3.org/nu/):

```HTML
<a href="#en">
  <input inert type="radio">
</a>
```

<br><br>

The HTML spec explains about [`a`](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element): _there must be no interactive content descendant_

But [`inert`](https://html.spec.whatwg.org/multipage/interaction.html#the-inert-attribute) means non-interactive: _Content in an inert subtree will not be perceivable by all users, or interactive_

**Comment by scottaohara** [comment]
what's the point of even trying to do this?

is this even a legitimate use of the inert attribute?  re: this passage from the spec about inert:
>Authors should not specify elements as inert unless the content they represent are also visually obscured in some way. In most cases, authors should not specify the [inert](https://html.spec.whatwg.org/multipage/interaction.html#inert) attribute on individual form controls. In these instances, the [disabled](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-disabled) attribute is probably more appropriate.

**Comment by tkrotoff** [comment]
> what's the point of even trying to do this?

That's another discussion but here a use case if you are interested: https://github.com/saadeghi/daisyui/discussions/4205

**Comment by sideshowbarker** [comment]
> That's another discussion but here a use case if you are interested: [saadeghi/daisyui#4205](https://github.com/saadeghi/daisyui/discussions/4205)

I’m just chiming in here to note that I’ve now read that, but on this issue — and in general on any other ARIA-related or accessibility-markup-related issues — I defer to @scottaohara’s judgment about whether there’s any problem that might require any fix to the checker.

But personally, based on just a quick look, all I can see is that once the `inert` attribute is removed from that `input`, it would certainly not be conforming. So it’s hard for me to see how many developers would want to mark up an `input` element as a child of an `a` element under any circumstances.

But that said, I haven’t spent any time thinking about it (and still won’t be, unless Scott or somebody tells me I need to).

**Comment by scottaohara** [comment]
Amazons markup/use of aria is very deliberate and using a radio-looking icon as an indicator to represent the aria-current state of the hyperlink. Nesting a radio button is not the right call here.

**Comment by tkrotoff** [comment]
I see, thx for your replies.

For example, it would have been an easy/dirty/naive way to style a link as a button (it's very common to have a link styled as a button):

```HTML
<a href="https://google.com">
  <button type="button" inert>This is a link "styled" as a button</button>
</a>
```

<img width="221" height="34" alt="Image" src="https://github.com/user-attachments/assets/408e8763-667e-415f-9e6e-e0afed8151d5" />

<br><br>

`<input>` and `<button>` are not the only elements to not accept `<a>` as parent, maybe [in this list](https://stackoverflow.com/a/6393863/990356) there are some "valid" use-cases with `inert`:

> you can nest any elements inside an `<a>` except the following:
> - `<a>`
> - `<audio>` (if the *controls* attribute is present)
> - `<button>`
> - `<details>`
> - `<embed>`
> - `<iframe>`
> - `<img>` (if the *usemap* attribute is present)
> - `<input>` (if the *type* attribute is not in the *hidden* state)
> - `<keygen>`
> - `<label>`
> - `<menu>` (if the *type* attribute is in the *toolbar* state)
> - `<object>` (if the *usemap* attribute is present)
> - `<select>`
> - `<textarea>`
> - `<video>` (if the *controls* attribute is present)

**Comment by scottaohara** [comment]
but if an element is marked as inert - it's hidden from the accessibility tree.  your example would result in a link that visually had text, but would be communicated to the browser's accessibility tree as if it didnt have any.

so yea, easy/dirty/naive - and also invalid and would result in accessibility errors.

**Comment by sideshowbarker** [comment]
Going ahead and closing this, based on the discussion
